js/bitmex.js

```diff
@@ -256,139 +421,155 @@ module.exports = class bitmex extends Exchange {
+            const typ = this.safeString (market, 'typ'); // type definitions at: https://www.bitmex.com/api/explorer/#!/Instrument/Instrument_get
+            const baseId = this.currencyIdFromAssetName (this.safeString (market, 'underlying'));
+            const quoteId = this.currencyIdFromAssetName (this.safeString (market, 'quoteCurrency'));
+            const settleId = this.currencyIdFromAssetName (this.safeString (market, 'settlCurrency', ''));
             const base = this.safeCurrencyCode (baseId);
             const quote = this.safeCurrencyCode (quoteId);
             const settle = this.safeCurrencyCode (settleId);
-            const basequote = baseId + quoteId;
-            const swap = (id === basequote);
             // 'positionCurrency' may be empty ("", as Bitmex currently returns for ETHUSD)
             // so let's take the settlCurrency first and then adjust if needed
             let type = undefined;
+            let spot = false;
             let future = false;
+            let swap = false;
             let prediction = false;
             let index = false;
-            let symbol = base + '/' + quote + ':' + settle;
+            let symbol = undefined;
             const expiryDatetime = this.safeString (market, 'expiry');
             const expiry = this.parse8601 (expiryDatetime);
             const inverse = this.safeValue (market, 'isInverse');
             const status = this.safeString (market, 'state');
             let active = status !== 'Unlisted';
-            if (swap) {
+            // types defined here: https://www.bitmex.com/api/explorer/#!/Instrument/Instrument_get
+            if (typ === 'FFWCSX' || typ === 'FFWCSF') {
                 type = 'swap';
+                swap = true;
+                symbol = base + '/' + quote + ':' + settle;
+            } else if (typ === 'IFXXXP') {
+                type = 'spot';
+                spot = true;
+                symbol = base + '/' + quote;
+            } else if (typ === 'FFCCSX') {
+                future = true;
+                type = 'future';
+                symbol = base + '/' + quote + ':' + settle + '-' + this.yymmdd (expiry);
             } else if (id.indexOf ('B_') >= 0) {
                 prediction = true;
                 type = 'prediction';
                 symbol = id;
-            } else if (expiry !== undefined) {
-                future = true;
-                type = 'future';
-                symbol = symbol + '-' + this.yymmdd (expiry);
             } else {
                 index = true;
                 type = 'index';
                 symbol = id;
                 active = false;
             }
+            const lotSize = this.safeString (market, 'lotSize');
+            let precisionAmount = undefined;
+            if (spot) {
+                precisionAmount = this.parseNumber (Precise.stringDiv ('1', lotSize));
+            } else {
+                precisionAmount = this.parseNumber (lotSize);
```

Comment:
@ttodua, currently for swap markets we have these huge "precisions" which does not seem very unified tbh.

For instance, in this case, the user would have to place an order with an amount >= 1000, and ok it might be the current implementation but if we want to "unify" it does not feel right, wdyt?

![image](https://user-images.githubusercontent.com/43336371/208096111-0cfff5d0-92a4-44a9-9c57-9c6a263c7509.png)

Reply:
ah, i've not touched them, my bad.
i will get into them to find out how existing implementation does that and how to solve.

Reply:
Carlos, the problems were more deep , that needed to restructure / redefine many things for future & swaps (which I hadnt touched till date), so will need  a bit more. will be ready on monday.

Reply:
@ttodua sure sure no rush 🙁